packages/common/src/utils/metricsExplorer.ts

```diff
@@ -236,19 +242,33 @@ export const getMetricExplorerDataPointsWithCompare = (
         comparison.type === MetricExplorerComparison.PREVIOUS_PERIOD
             ? metricId
             : getItemId({
-                  table: comparison.metricTable,
-                  name: comparison.metricName,
+                  table: comparison.metric.table,
+                  name: comparison.metric.name,
               });
 
+    let comparisonMetricLabel = null;
+    if (comparison.type === MetricExplorerComparison.DIFFERENT_METRIC) {
+        comparisonMetricLabel =
+            comparison.metric.label ?? comparison.metric.name;
+    } else if (comparison.type === MetricExplorerComparison.PREVIOUS_PERIOD) {
+        comparisonMetricLabel = 'Previous Period';
+    }
+
     return Array.from(dates).map((date) => ({
         date: new Date(date),
-        metric: parseMetricValue(
-            groupByMetricRows[date]?.[0]?.[metricId]?.value.raw,
-        ),
-        compareMetric: parseMetricValue(
-            offsetGroupByCompareMetricRows[date]?.[0]?.[compareMetricId]?.value
-                .raw,
-        ),
+        metric: {
+            value: parseMetricValue(
+                groupByMetricRows[date]?.[0]?.[metricId]?.value.raw,
+            ),
+            label: metric.label ?? metric.name,
+        },
+        compareMetric: {
+            value: parseMetricValue(
+                offsetGroupByCompareMetricRows[date]?.[0]?.[compareMetricId]
+                    ?.value.raw,
+            ),
+            label: comparisonMetricLabel,
```

Comment:
here is the label - then the tooltip just consumes the value without having to do any finds or extra processing on the FE